crates/libs/core/src/types/client/health.rs

```diff
@@ -0,0 +1,947 @@
+    fn from(value: &FABRIC_APPLICATION_HEALTH_REPORT) -> Self {
+        Self {
+            application_name: WString::from_wide(unsafe {
+                PCWSTR::from_raw(value.ApplicationName.0).as_wide()
+            }),
+            health_information: HealthInformation::from(unsafe {
+                value.HealthInformation.as_ref().unwrap()
+            }),
+        }
+    }
+}
+
+/// FABRIC_DEPLOYED_APPLICATION_HEALTH_REPORT
+#[derive(Debug, Clone)]
+pub struct DeployedApplicationHealthReport {
+    pub application_name: WString,
+    pub node_name: WString,
+    pub health_information: HealthInformation,
+    // TODO: Implement reserved fields
+    // pub reserved: *mut std::ffi::c_void,
+}
+
+impl From<&DeployedApplicationHealthReport> for FABRIC_DEPLOYED_APPLICATION_HEALTH_REPORT {
+    fn from(value: &DeployedApplicationHealthReport) -> Self {
+        Self {
+            ApplicationName: FABRIC_URI(value.application_name.as_ptr() as *mut u16),
+            NodeName: value.node_name.as_pcwstr(),
+            HealthInformation: &FABRIC_HEALTH_INFORMATION::from(&value.health_information),
+            Reserved: std::ptr::null_mut(),
+        }
+    }
+}
+
+impl From<&FABRIC_DEPLOYED_APPLICATION_HEALTH_REPORT> for DeployedApplicationHealthReport {
+    fn from(value: &FABRIC_DEPLOYED_APPLICATION_HEALTH_REPORT) -> Self {
+        Self {
+            application_name: WString::from_wide(unsafe {
+                PCWSTR::from_raw(value.ApplicationName.0).as_wide()
+            }),
+            node_name: WString::from_wide(unsafe { value.NodeName.as_wide() }),
+            health_information: HealthInformation::from(unsafe {
+                value.HealthInformation.as_ref().unwrap()
+            }),
+        }
+    }
+}
+
+/// FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT
+#[derive(Debug, Clone)]
+pub struct DeployedServicePackageHealthReport {
+    pub application_name: WString,
+    pub service_manifest_name: WString,
+    pub node_name: WString,
+    pub health_information: HealthInformation,
+    // TODO: Implement reserved fields
+    // pub reserved: *mut std::ffi::c_void,
+}
+
+impl From<&DeployedServicePackageHealthReport> for FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT {
+    fn from(value: &DeployedServicePackageHealthReport) -> Self {
+        Self {
+            ApplicationName: FABRIC_URI(value.application_name.as_ptr() as *mut u16),
+            ServiceManifestName: value.service_manifest_name.as_pcwstr(),
+            NodeName: value.node_name.as_pcwstr(),
+            HealthInformation: &FABRIC_HEALTH_INFORMATION::from(&value.health_information),
+            Reserved: std::ptr::null_mut(),
+        }
+    }
+}
+
+impl From<&FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT> for DeployedServicePackageHealthReport {
+    fn from(value: &FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT) -> Self {
+        Self {
+            application_name: WString::from_wide(unsafe {
```

Comment:
WStringWrap can do conversions directly: https://github.com/Azure/service-fabric-rs/blob/f8cf77f3290b2c7a61aaecd8a7d36378898d1eca/crates/libs/core/src/strings.rs#L59
Maybe WStringWrap needs to be cleaned up or merge with Wstring.

Reply:
Changed to use `WStringWrap` for safe conversion